api/docs/mapstruct.md

```diff
@@ -0,0 +1,608 @@
+  EmployeeModel toModel(EmployeeDbEntity employeeDbEntity);
+                ^
+/Users/jaycarlton/repos/workbench/api/build/generated/sources/annotationProcessor/java/main/org/pmiops/workbench/utils/mappers/examples/EmployeeMapperImpl.java:12: error: enum types may not be instantiated
+    private final Department department = new Department();
+                                          ^
+/Users/jaycarlton/repos/workbench/api/build/generated/sources/annotationProcessor/java/main/org/pmiops/workbench/utils/mappers/examples/EmployeeMapperImpl.java:20: error: clone() has protected access in Enum
+        EmployeeModel employeeModel = department.clone();
+                                                ^
+/Users/jaycarlton/repos/workbench/api/build/generated/sources/annotationProcessor/java/main/org/pmiops/workbench/utils/mappers/examples/EmployeeMapperImpl.java:20: error: incompatible types: Object cannot be converted to EmployeeModel
+        EmployeeModel employeeModel = department.clone();
+                                                      ^
+```
+
+Fine, we can do it the less slick but more reliable way:
+
+```java
+@Mapper(uses = SampleEnumMapper.class)
+public interface EmployeeMapper {
+
+  @Mapping(source = "employeeDbEntity", target = "name")
+  @Mapping(source = "employeeDbEntity.departmentCode", target = "department")
+  EmployeeModel toModel(EmployeeDbEntity employeeDbEntity);
+
+  default String toNickname(EmployeeDbEntity employeeDbEntity) {
+    return String.format("%s %s", employeeDbEntity.getFirstName(), employeeDbEntity.getLastName());
+  }
+}
+```
+
+where the new `SampleEnumMapper` looks like:
+```java
+@Mapper
+public interface SampleEnumMapper {
+  default Department toDepartment(int departmentCode) {
+    return Department.fromDepartmentCode(departmentCode);
+  }
+}
+```
+
+I'm sure there's a way to do this without creating another mapper class. Anyway, this gives us 
+```java
+public class EmployeeMapperImpl implements EmployeeMapper {
+
+    private final SampleEnumMapper sampleEnumMapper = Mappers.getMapper( SampleEnumMapper.class );
+
+    @Override
+    public EmployeeModel toModel(EmployeeDbEntity employeeDbEntity) {
+        if ( employeeDbEntity == null ) {
+            return null;
+        }
+
+        EmployeeModel employeeModel = new EmployeeModel();
+
+        employeeModel.setName( toNickname( employeeDbEntity ) );
+        employeeModel.setDepartment( sampleEnumMapper.toDepartment( employeeDbEntity.getDepartmentCode() ) );
+        employeeModel.setAddress( employeeDbEntity.getAddress() );
+
+        return employeeModel;
+    }
+}
+```
+Our `@Mapping` instruction let MapStruct know which source property (an int) to match to the target
+property (`Department`). There was only one such method, so it was selected.
+
+To see what happens when there's ambiguity, add another method to `SampleEnumMapper`:
```

Comment:
Add how to resolve ambiguity here.  See `qualifiedByName` in `VerifiedInstitutionalAffiliationMapper`